### PR TITLE
[urijs] Remove unnecessary JQuery dependency per #35453

### PR DESCRIPTION
**Pull request by TeamworkGuy2**
Removing unnecessary JQuery import per request #35453 

Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  https://www.npmjs.com/package/urijs and https://github.com/DefinitelyTyped/DefinitelyTyped/issues/35453
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/urijs/index.d.ts [hunk 1]
```diff
@@ -210,11 +192,11 @@ declare namespace uri {
         encode(str: string): string;
         encodeQuery(qry: string): string;
         encodeReserved(str: string): string;
-        expand(template: string, vals: Object): URI;
+        expand(template: string, vals: object): URI;
 
         iso8859(): void;
 
-        joinPaths(...paths: (string | URI)[]): URI;
+        joinPaths(...paths: Array<string | URI>): URI;
```
Comment (ljqx) [inline types/urijs/index.d.ts]: why in `hasSearch` prefer `Type[]` and here prefer `Array<Type>`?

**Review by ljqx — CHANGES_REQUESTED** [review]

File: types/urijs/index.d.ts [hunk 2]
```diff
@@ -1,45 +1,38 @@
-// Type definitions for URI.js 1.19.1
-// Project: https://github.com/medialize/URI.js
+// Type definitions for URI.js 1.19
+// Project: http://medialize.github.io/uri.js
```
[the remark this answers is not in the document]
Comment (TeamworkGuy2) [reply types/urijs/index.d.ts]: URLs are case insensitive and tslint was complaining the the URL didn't match the npm package, so this fixed that tslint error.

File: types/urijs/index.d.ts [hunk 3]
```diff
@@ -284,16 +265,16 @@ interface JQuery {
 }
```
[the remark this answers is not in the document]
Comment (TeamworkGuy2) [reply types/urijs/index.d.ts]: Yes, if your project also includes the jquery type definitions, the JQuery interfaces in both will be merged.

File: types/urijs/index.d.ts (hunk 1 above)
Comment (TeamworkGuy2) [reply types/urijs/index.d.ts]: tslint error, complex types must use Array<...>

File: types/urijs/urijs-tests.ts [hunk 4]
```diff
@@ -192,14 +190,18 @@ URI.build({
...
 From: https://medialize.github.io/URI.js/docs.html#static-parse
 */
-var parts = URI.parse('mailto:mail@example.org');
+let parts = URI.parse('mailto:mail@example.org');
 parts.path === "mail@example.org";
 parts.preventInvalidHostname = false;
 parts.protocol === "mailto";
 parts.urn === true;
+
+function test(a: any, b: any): boolean {
```
[the remark this answers is not in the document]
Comment (TeamworkGuy2) [reply types/urijs/urijs-tests.ts]: Updated, new commit!

**Comment by TeamworkGuy2** [comment]
@ljqx thanks for the review, see comments.

File: types/urijs/index.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ljqx) [reply types/urijs/index.d.ts]: but this url is not working...

Comment (ljqx) [inline types/urijs/index.d.ts]: Actually this URL is case-sensitive, only when you change the `uri` to upper case you can go to the document site

**Review by ljqx — CHANGES_REQUESTED** [review]

**Review by ljqx — CHANGES_REQUESTED** [review]
Fix the document url then I think it's OK

File: types/urijs/index.d.ts (hunk 2 above)
Comment (TeamworkGuy2) [reply types/urijs/index.d.ts]: done

**Comment by TeamworkGuy2** [comment]
@ljqx done

**Review by ljqx — APPROVED** [review]

**Pull request merged**